notebooks/kkl15.jl

```diff
@@ -0,0 +1,1612 @@
+
+We specify three contrasts for the four-level factor CTR that are derived from spatial, object-based, and attractor-like features of attention.  They map onto sequential differences between appropriately ordered factor levels. Replicating Kliegl et al. (2011), the attraction effect was not significant as a fixed effect, but yielded a highly reliable variance component (VC; i.e., reliable individual differences in positive and negative attraction effects cancel the fixed effect). Moreover, these individual differences in the attraction effect were negatively correlated with those in the spatial effect.
+
+A few years after the publication of Kliegl et al. (2011), we determined that the theoretically critical correlation parameter (CP) between the spatial effect and the attraction effect was the source of a non-singular LMM in that paper. The present study served the purpose to estimate this parameter with a larger sample and a wider variety of experimental conditions. Therefore, the code in this script is identical to `kwdyz.jl` for now. We may return later to illustrate issues of model selection when additional within-subject manipulations relating to the orientiation of the rectangle cues are modelled in the random effect structure.  
+
+The analyses were originally reported in the parsimonious mixed-model paper [(Bates et al., 2015)](https://arxiv.org/abs/1506.04967). Data and R scripts are also available in [R-package RePsychLing](https://github.com/dmbates/RePsychLing/tree/master/data/) (Baayen et al., 2014). In this script and in `kwdyz11.jl` we provide the corresponding analyses with _MixedModels.jl_.
+
+## Packages
+"""
+
+# ╔═╡ 539399d9-1b68-4d90-8e63-ff5f2674c494
+md" ## Read data, compute and plot means"
+
+# ╔═╡ 47e078b0-41ab-4186-8680-aff3a79ea3c7
+begin
+	dat1 = DataFrame(Arrow.Table("./data/kkl15.arrow"))
+	dat1 = select(dat1, :subj => :Subj, :tar => :CTR, :rt)
+	dat1.CTR = categorical(dat1.CTR)
+	levels!(dat1.CTR, ["val", "sod", "dos", "dod"])
+	describe(dat1)
+end
+
+# ╔═╡ 90b45536-4397-418a-8bea-887a6d5bdff9
+begin
+	# Descriptive statistics
+	## Subject level
+	dat_subj = combine(groupby(dat1, [:Subj, :CTR]), 
+					   :rt => length => :n, 
+		               :rt => mean => :rt_m,
+		               :rt  => std => :rt_sd);
+	## Condition level
+	dat_cond = combine(groupby(dat_subj, [:CTR]), 
+					   :n => length => :N, 
+		               :rt_m => mean => :rt_M, 
+		               :rt_m  => std => :rt_SD, 
+                       :rt_m => (x -> std(x)/sqrt(length(x))) => :rt_SE)
+end
+
+# ╔═╡ dd297971-1728-48d8-b73e-94d87555355e
+begin
+	geom = visual(BoxPlot, layout_x = 1) + visual(Violin, layout_x = 2)
```

Comment:
Did you intend for this plot to have both a boxplot and a violin plot?

Would it be okay with you if I transposed the plot? I keep telling students that most plots of a numeric response by a categorical covariate should have the response on the horizontal axis.  Most times you can shrink the vertical axis without losing any information because it is just four categories.

Reply:
Actually, my preference would be a dotplot(). I was playing with boxplot() and violinploit() and did not really like either.    I agree with the coord_flip().  Can you implement this? I don't know how to do this with AoG. (Also I did  not try it, but if there is a dotplot(), feel free to exchange it.

Reply:
PS: I am in a meeting and cannot act myself currently.

Reply:
Okay, I will experiment,  It may take a while. (But I'm sure it will be worth the effort.)

Reply:
👍

Reply:
So I am starting to learn more about the data and only now realized that there are 147 subjects.  The `Subj` variable has 86 levels but they go from "1" to "9", lexicographically.  I have changed those to "S001" to "S147" in a modified version of the notebook for you to check.  I also added a CairoMakie boxplot with the horizontal orientation and a comparative densityplot of all the response times.  These are just suggestions.

Also, I found out that there are 24 columns in the Arrow table.  Are they all necessary or are some of them derived columns?  It appears that many of them are columns extracted from a model matrix.  Storing those as separate columns is a table is a practice I would strongly discourage.